tfjs-core/src/ops/browser.ts

```diff
@@ -371,4 +382,46 @@ export async function toPixels(
   return bytes;
 }
 
+/**
+ * Draws a `tf.Tensor` to a canvas.
+ *
+ * When the dtype of the input is 'float32', we assume values in the range
+ * [0-1]. Otherwise, when input is 'int32', we assume values in the range
+ * [0-255].
+ *
+ * @param image A rank-2 tensor with shape `[height, width]`, or a rank-3 tensor
+ * of shape `[height, width, numChannels]`. If rank-2, draws grayscale. If
+ * rank-3, must have depth of 1, 3 or 4. When depth of 1, draws
+ * grayscale. When depth of 3, we draw with the first three components of
+ * the depth dimension corresponding to r, g, b and alpha would be 1 (opaque) or
+ * customized. When depth of 4, all four components of the depth dimension
+ * correspond to r, g, b, a.
+ * @param canvas The canvas to draw to.
+ * @param contextOptions A object with options to get the context from the
+ *     canvas. If the canvas has created context, contextOptions would not make
+ * effects.
+ * @param drawOptions A object of options to customize drawing.
+ *
+ * @doc {heading: 'Browser', namespace: 'browser'}
+ */
+export function draw(
+    image: Tensor2D|Tensor3D|TensorLike, canvas: HTMLCanvasElement,
+    contextOptions?: ContextOptions, drawOptions?: DrawOptions): void {
```

Comment:
Is it simpler to merge `contextOptions` and `drawOptions` into `CanvasOptions`/`CanvasConfiguration`? For example, in webgpu, we have [GPUCanvasConfiguration](https://www.w3.org/TR/webgpu/#dictdef-gpucanvasconfiguration)
```
export interface CanvasOptions {
  /**
   * Optional. If it is not set, it would be variable based on the current
   * backend.
   */
  contextType?: string;
  /**
   * Optional.
   */
  contextAttributes?: WebGLContextAttributes;

  /**
   * Optional. A number in range [0-1]. If the image is a 2D tensor or a 3D
   * tensor with 1 or 3 channels, the alpha channels would set as its value;
   * otherwise, it would not make effects.
   */
  alpha?: number;
}

Reply:
[GPUCanvasConfiguration](https://www.w3.org/TR/webgpu/#dictdef-gpucanvasconfiguration) has alphaMode, but it is different from the alpha here.

I mean either contextOptions or [GPUCanvasConfiguration](https://www.w3.org/TR/webgpu/#dictdef-gpucanvasconfiguration) are describing the setting of canvas or context. However, drawOptions is to customize the kernel computations. Separating them might be more clearer for users?

Also, do you think the name and the definition of `drawOptions` makes sense for WebGPU?

Reply:
I think it's ok to custom `alpha`. I don't have good candidate name for `drawOptions`. Leave it open if other reviewers have better suggestion or just keep the current name.

Reply:
Sorry, I mean `contextOptions` (updated my comment), does it makes senses for WebGPU? Should I add more optional fields, like [GPUCanvasConfiguration](https://www.w3.org/TR/webgpu/#dictdef-gpucanvasconfiguration)? I intend to wrap all canvas and context related settings into it.

Reply:
I agree that merging two options into one can cause some confusion in some overlapping but different purpose params like `alpha`. But I feel a little bit concerned when I see your example usage code:
```js
tf.browser.draw(res, canvasElem, undefined, {alpha: 0.5});
```
From the user's perspective it is hard to tell whether the first one or second one is drawOptions. We may hurt the readability and usability of this API. Just my personal concerns, throw it out for more discussion.

Also, if we choose to release it as two separate options params, consider to make it:
```js
function draw(
    image: Tensor2D|Tensor3D|TensorLike, canvas: HTMLCanvasElement, drawOptions?: DrawOptions, contextOptions?: ContextOptions)
```
if you think customizing `drawOptions` is a normal user feature but specifying  `contextOptions` is more like an feature for advanced users, swapping the order may be more easy to use since most of the users can just do `tf.browser.draw(res, canvas, { alpha: 1, ...});`

Reply:
1. Since we would add `GPUCanvasConfiguration ` to `CanvasOptions`, I renamed `ContextOptions` to `CanvasOptions`.
2. For usage of the API with two objects, if users follow the coding style, it would be:
```js
const canvasOptions = {};
const drawOptions = {alpha: 0.5}
tf.browser.draw(res, canvasElem, canvasOptions, drawOptions);
```
3. I am open to suggestions for whether we want to merge the two options as `Options: {CanvasOptions, DrawOptions}`.

5.

Reply:
As collected feedbacks from you and WebML channel, get most votes for unified option object for this. Updated.